index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
Sure.  It's helpful when you have some transformation you want to do to a stream in certain cases.

E.g. say we want to `concat` some warning messages in a UI if a form field is empty

```
form.map( 
  when(
     view( fields, any( isEmpty) )
     ,over( warnings, concat(['Please fill in all form fields'])
   )
)
```

As we emit form objects in our stream, we'll conditionally add validation messages if necessary.

Notice there's no need for an explicit else case.  The else case is to simply leave the structure as is.  You could use `ifElse( ..., identity)`, but it feels cumbersome.